src/Chapter2.hs

```diff
@@ -841,7 +859,11 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n l
+  | n < 0 = []
+  | otherwise = take len (drop n (cycle l))
+  where len = length l
```

Comment:
This implementation is also what we expect and it's efficient enough :+1: `length` is slow for lists, but here not much you can do to avoid it.

There exist an alternative lazy solution using `zip`, but it's quite advanced (in terms of understanding what's going on), and relies on laziness

* https://stackoverflow.com/questions/16378773/rotate-a-list-in-haskell